userspace/libscap/scap_procs.c

```diff
@@ -403,6 +403,72 @@ static int32_t scap_proc_fill_flimit(scap_t *handle, uint64_t tid, struct scap_t
 }
 #endif
 
+int32_t scap_proc_fill_overlayfs_root(scap_t *handle, struct scap_threadinfo *tinfo, const char *procdirname)
+{
+	FILE *mounts_file;
+	char mounts_path[SCAP_MAX_PATH_SIZE];
+	char line[SCAP_MAX_ARGS_SIZE];
+
+	tinfo->overlayfs_root[0] = '\0';
+
+	snprintf(mounts_path, sizeof(mounts_path), "%smounts", procdirname);
+	mounts_file = fopen(mounts_path, "r");
+	if(!mounts_file)
+	{
+		return SCAP_FAILURE;
+	}
+
+	if(!fgets(line, sizeof(line), mounts_file))
+	{
+		fclose(mounts_file);
+		return SCAP_FAILURE;
+	}
+
+	fclose(mounts_file);
+
+	if(strstr(line, "overlay / overlay"))
+	{
+		char *start, *end;
+
+		//
+		//  Get the lower dir
+		//
+		start = strstr(line, "lowerdir=") + sizeof("lowerdir=") - 1;
+		if(!start)
+		{
+			return SCAP_FAILURE;
+		}
+
+		end = strstr(start, ",");
+		if(!end)
+		{
+			return;
+		}
+		*end = '\0';
+
+		// Docker lowerdir entry has this shape:
+		// lowerdir=/var/lib/docker/overlay2/l/KAHXLZBU2SRCXTUAGNXYDELYRI:/var/lib/docker/overlay2/l/Y35R5SXA45GJIQFSMFK3HDXM2D
+		// To get the actual lowerdir we have to use tha path right after the ':', so we must handle this as a special case
+		// Containerd does not have this problem
+		// todo: test with cri-o and podman
+		if(strstr(start, "docker"))
```

Comment:
This is an ultra-nitpick, but maybe check against `/docker/`? Somehow I feel `docker` could find its way onto some other part of the string and mess with this check.